Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -236,6 +328,81 @@ lemma norm_fourierPowSMulRight_le (f : V → E) (v : V) (n : ℕ) :
+lemma norm_iteratedFDeriv_fourierPowSMulRight
+    {f : V → E} {K : ℕ∞} {C : ℝ} (hf : ContDiff ℝ K f) {n : ℕ} {k : ℕ} (hk : k ≤ K)
+    {v : V} (hv : ∀ i ≤ k, ∀ j ≤ n, ‖v‖ ^ j * ‖iteratedFDeriv ℝ i f v‖ ≤ C) :
+    ‖iteratedFDeriv ℝ k (fun v ↦ fourierPowSMulRight L f v n) v‖ ≤
+      (2 * π) ^ n * (2 * n + 2) ^ k * ‖L‖ ^ n * C := by
+  /- We write `fourierPowSMulRight L f v n` as a composition of bilinear and multilinear maps,
+  thanks to `fourierPowSMulRight_eq_comp`, and then we control the iterated derivatives of these
+  thanks to general bounds on derivatives of bilinear and multilinear maps. -/
+  simp_rw [fourierPowSMulRight_eq_comp]
+  let T : (W →L[ℝ] ℝ) [×n]→L[ℝ] (W [×n]→L[ℝ] ℝ) :=
+    compContinuousLinearMapLRight (ContinuousMultilinearMap.mkPiAlgebra ℝ (Fin n) ℝ)
+  have I₁ m : ‖iteratedFDeriv ℝ m T (fun _ ↦ L v)‖ ≤
+      Nat.descFactorial n m * 1 * (‖L‖ * ‖v‖) ^ (n - m) := by
+    have : ‖T‖ ≤ 1 := by
+      apply (norm_compContinuousLinearMapLRight_le _ _).trans
+      simp only [norm_mkPiAlgebra, le_refl]
+    apply (ContinuousMultilinearMap.norm_iteratedFDeriv_le _ _ _).trans
+    simp only [Fintype.card_fin]
+    gcongr
+    refine (pi_norm_le_iff_of_nonneg (by positivity)).mpr (fun _i ↦ ?_)
+    exact ContinuousLinearMap.le_opNorm _ _
+  have I₂ m : ‖iteratedFDeriv ℝ m (T ∘ (ContinuousLinearMap.pi (fun (_i : Fin n) ↦ L))) v‖ ≤
+      (Nat.descFactorial n m * 1 * (‖L‖ * ‖v‖) ^ (n - m)) * ‖L‖ ^ m := by
+    rw [ContinuousLinearMap.iteratedFDeriv_comp_right _ (ContinuousMultilinearMap.contDiff _)
+      _ le_top]
+    apply (norm_compContinuousLinearMap_le _ _).trans
+    simp only [Finset.prod_const, Finset.card_fin]
+    gcongr
+    · exact I₁ m
+    · exact ContinuousLinearMap.norm_pi_le_of_le (fun _i ↦ le_rfl) (norm_nonneg _)
+  have I₃ m : ‖iteratedFDeriv ℝ m (T ∘ (ContinuousLinearMap.pi (fun (_i : Fin n) ↦ L))) v‖ ≤
+      Nat.descFactorial n m * ‖L‖ ^ n * ‖v‖ ^ (n - m) := by
+    apply (I₂ m).trans (le_of_eq _)
+    rcases le_or_lt m n with hm|hm
+    · have : ‖L‖ ^ n = ‖L‖ ^ (m + (n - m)) := by congr; omega
+      rw [this, pow_add]
+      ring
+    · simp [Nat.descFactorial_eq_zero_iff_lt.mpr hm]
+  have A : ContDiff ℝ K fun y ↦ T fun _ ↦ L y := by
+    apply (ContinuousMultilinearMap.contDiff _).comp
+    exact contDiff_pi.2 (fun _ ↦ L.contDiff)
+  rw [iteratedFDeriv_const_smul_apply', norm_smul (β := V [×k]→L[ℝ] (W [×n]→L[ℝ] E))]; swap
+  · exact (smulRightL ℝ (fun (_ : Fin n) ↦ W) E).isBoundedBilinearMap.contDiff.comp₂ (A.of_le hk)
+      (hf.of_le hk)
+  simp only [norm_pow, norm_neg, norm_mul, RCLike.norm_ofNat, Complex.norm_eq_abs, abs_ofReal,
+    _root_.abs_of_nonneg pi_nonneg, abs_I, mul_one, mul_assoc]
+  gcongr
+  apply (ContinuousLinearMap.norm_iteratedFDeriv_le_of_bilinear_of_le_one _ A hf _
+    hk ContinuousMultilinearMap.norm_smulRightL_le).trans
+  calc
+  ∑ i in Finset.range (k + 1),
+    Nat.choose k i * ‖iteratedFDeriv ℝ i (fun (y : V) ↦ T (fun _ ↦ L y)) v‖ *
```

Comment:
```suggestion
    k.choose i * ‖iteratedFDeriv ℝ i (fun (y : V) ↦ T (fun _ ↦ L y)) v‖ *
```
(saves a few bytes, but more importantly better matches usual informal notation) -- same in a few other places here.